### PR TITLE
First pass at a development environment

### PR DESCRIPTION
**Pull request by sagepe**
Adds an initial development environment based on Docker Compose that should be good enough for most front-end changes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM mysocietyorg/apache-php-fpm:stretch
+LABEL maintainer="sysadmin@mysociety.org"
+
+RUN apt-get update \
+      && apt-get install -y \
+          libconvert-base32-perl \
```
Comment (dracos) [inline Dockerfile:6]: Can a Dockerfile not read a list from a file somehow? :) COPY conf/packages inside and then run it on that?

Comment (sagepe) [reply Dockerfile:6]: Yes, we could, as long as the packages file is a simple list of package, we could `COPY` it in then do something like `RUN xargs -a packages apt-get install -y --no-install-recommends`.

Comment (sagepe) [reply Dockerfile]: Added in [3846d20](https://github.com/mysociety/writetothem/pull/464/commits/3846d2098719cfdefec380eb386b27d225a2dc01)

**Comment by zarino** [comment]
Just tried this out. Searched for my postcode from the homepage, got this error:

Screenshot_2021-02-09 WriteToThem

Just me?

**Comment by dracos** [comment]
It doesn't handle anything past display, I'm afraid, there's no connection to a backend system containing rep information.

**Comment by sagepe** [comment]
Yes, as per @dracos comment this is just an initial environment hopefully good enough for most front-end changes (see [this commit message](https://github.com/mysociety/writetothem/pull/464/commits/f17eedf4eef54b2bc996591f88cd94a4eb0fc2a0)).

I'd like to get it working beyond this at some stage, but didn't want to block @wrightmartin's contact page task while getting a load of back-end stuff working.

The quickest way to get a more fully functional system may be to look at running it on the VPN so it can access the internal services; we'd still need a daemon container and one or two other bits, but that might be the quickest win.

**Comment by zarino** [comment]
No worries! 👍

**Review by zarino — APPROVED** [review]
Been using this for https://github.com/mysociety/writetothem/pull/467. Works fine for changes to static pages like the Help pages, but doesn’t help if you want to change any of the screens in the main request flow, because the postcode and MP lookups don’t work. Probably worth merging, regardless.

**Review by dracos — COMMENTED** [review]
Do we need the giant php.ini file? Can we provide just an override for the things that it needs, or at least document what it's doing differently? From a local diff, it looks like it's included in order to set short_open_tag to On, add ~E_NOTICE to error_reporting, and set mail.add_x_header to On.

**Comment by dracos** [comment]
Or for now you could just put those three in a conf.d ini file inside /etc/php..., would that have the same effect?

**Comment by sagepe** [comment]
> Do we need the giant php.ini file? 

No, you are right. We simply need `short_open_tag` for Stretch; I was experimenting with Buster at one stage, which is where the `E-NOTICE` change came in as our code generates a few of these and things weren't loading right. We can revisit this later. I think must be where the `mail.add_x_header` crept in, too, I didn't change this myself - it must have come from a different source file.

> the first we could sort in the repo ... Or for now you could just put those three in a conf.d ini file inside /etc/php...

I've done the latter for now, although the former is probably something we should do in due course.

**Comment by sagepe** [comment]
I think we can merge this, I know it would be ideal to get a backend bit working but as noted it should be OK for front-end changes.

File: README.md [hunk 2]
```diff
@@ -10,6 +10,42 @@ built from the start to be more flexible and easier to use.
 
 ## Developing WriteToThem
 
+This repository contains a simple development environment based on Docker Compose.
+
+Currently, this environment can be used for makeing changes to the front-end of the
```
Comment (dracos) [inline README.md:15]: making

**Review by dracos — CHANGES_REQUESTED** [review]
There's one typo in the readme is all :)

File: README.md (hunk 2 above)
Comment (sagepe) [reply README.md]: Fix has been squashed in.

**Review by dracos — APPROVED** [review]
Could probably squash the use-conf-packages commit in unless you didn't want to :)

**Pull request merged**